metrics-core/src/test/java/com/codahale/metrics/Slf4jReporterTest.java

```diff
@@ -4,66 +4,89 @@
 import org.slf4j.Logger;
 import org.slf4j.Marker;
 
+import java.util.EnumSet;
+import java.util.Set;
 import java.util.SortedMap;
 import java.util.TreeMap;
 import java.util.concurrent.TimeUnit;
 
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static com.codahale.metrics.MetricAttribute.*;
+import static org.mockito.Mockito.*;
 
 public class Slf4jReporterTest {
+
     private final Logger logger = mock(Logger.class);
     private final Marker marker = mock(Marker.class);
     private final MetricRegistry registry = mock(MetricRegistry.class);
-    private final Slf4jReporter infoReporter = Slf4jReporter.forRegistry(registry)
-            .outputTo(logger)
-            .markWith(marker)
-            .prefixedWith("prefix")
-            .convertRatesTo(TimeUnit.SECONDS)
-            .convertDurationsTo(TimeUnit.MILLISECONDS)
-            .withLoggingLevel(Slf4jReporter.LoggingLevel.INFO)
-            .filter(MetricFilter.ALL)
-            .build();
-
-    private final Slf4jReporter errorReporter = Slf4jReporter.forRegistry(registry)
-            .outputTo(logger)
-            .markWith(marker)
-            .convertRatesTo(TimeUnit.SECONDS)
-            .convertDurationsTo(TimeUnit.MILLISECONDS)
-            .withLoggingLevel(Slf4jReporter.LoggingLevel.ERROR)
-            .filter(MetricFilter.ALL)
-            .build();
+
+    private Set<MetricAttribute> disabledMetricAttributes = null;
```

Comment:
Instead of declaring disabledMetricAttributes as a field, I think we should it make a parameter to the `infoReporter` and `errorReporter` methods. Othwerise it's not clear how what's a relationship between it and the actual reporter.

Reply:
I like to use this pattern to override default values passed to the constructor of the main unit tested component. I think in the context of unit tests it makes sense since there is only 1 class that is supposed to be tested. In the context of Slf4jReporter, it should be clear that the only way of testing Slf4jReporter is to use its builder and that its builder takes an optional `Set<MetricAttribute> disabledMetricAttributes`.
This pattern becomes particularly handy when there might be more than 1 default value to overwrite for some tests. The alternatives would be to either invoke the full builder with all the proper default values set but this leads to code duplication; or to add more arguments to the `infoReporter` and `errorReporter` factory methods to overwrite the default value but this doesn't scale well when more than a couple of variables needs their default values overwritten and the factory method may end up having as many parameters as the builder...
I'll add a comment clarifying this is used to build the reporter in each test and can be overridden. I will also move at the top of the infoReporter and errorReporter methods to hep identify the relationship.